src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -33,6 +33,8 @@ public static RazorSyntaxTree Rewrite(RazorSyntaxTree syntaxTree, string tagHelp
 
         var diagnostics = CombineErrors(syntaxTree.Diagnostics, errorList).OrderBy(error => error.Span.AbsoluteIndex);
 
+        usedDescriptors = rewriter.UsedDescriptors;
```

Comment:
This is exposing a set as a list. That means the list is non-deterministic. Generally to be avoided in the compiler layer. 

Have we verified order isn't imporant here?

Reply:
I'm using a set because we expect there to be duplicates (for example if you use the same tag helper twice in one document) and we just want to know if it was used at all.

This is actually a good point though. The downstream change is treating as a list and considering order. Order actually isn't important for this check. Not sure I can construct a situation in which the order changes and files wouldn't need to reparse, but we should make the semantics clear. Will switch this to a set.